### ISSUE TITLE
Will nodeports in 2048 example conflict if multiple pods allocated in the same node?

### ISSUE THREAD

**Issue by JoHuang** [kubernetes-sigs/aws-alb-ingress-controller#1155]
I would like to try the nodeport / instance mode of this project.

My question is:
If I have 2 nodes, and I scale up the 2048 pod to 3, there is one node containing 2 pods.
Will the 2 pods have the same nodeport number and conflict each other?
Should I use instance mode in production?

Thanks,
Jo

**Comment by keyvhinng** [comment]
Hi @JoHuang, if you check this good article https://aws.amazon.com/blogs/opensource/kubernetes-ingress-aws-alb-ingress-controller, you will resolve many of your doubts.

TL;DR: A `NodePort` is created by service (Service type: NodePort), not by a pod.

A request reaching a `NodePort` in a node will be redirected to one of the PODs that belong to a `deployment` which was exposed as a `Service` (this service has a specific port in each node: `NodePort`)

In summary, if two pods belong to the same service and are in the same node, the requests will be balanced across these two pods

**Comment by M00nF1sh** [comment]
thanks @keyvhinng for the excellent answer 🤣 

@JoHuang Yes, you should use instance mode in production. IP mode have limitations around zero-downtime deployment currently.

closing this one, feel free to reopen.

**Comment by JoHuang** [comment]
Thanks for detail explanation.
